autotest/pyscripts/test_ogr_layer_algebra.py

```diff
@@ -0,0 +1,489 @@
+#!/usr/bin/env pytest
+###############################################################################
+# $Id$
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  To test the functionality of ogr_layer_algebra script
+# Author:   Radha Krishna Kavuluru <kssvrk@gmail.com>
+#
+###############################################################################
+# Copyright (c) 2022, Radha Krishna Kavuluru <kssvrk@gmail.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+import pytest
+import test_py_scripts
+
+from osgeo import ogr
+
```

Comment:
as ogr_layer_algebra algorithms require GEOS support, we must skip the tests if it is not available:

```suggestion

import ogrtest

###############################################################################
@pytest.fixture(autouse=True, scope="module")
def startup_and_cleanup():

    if not ogrtest.have_geos():
        pytest.skip()


```

Reply:
changed in https://github.com/kssvrk/gdal/commit/841f3fe4b409219623527f0a021c34eae6071697